CMakeLists.txt

```diff
@@ -56,21 +56,22 @@ macro(add_flag flag)
   add_cflag(${flag})
   add_cxxflag(${flag})
 endmacro()
-  
-# Set standard version for compiler.
-add_cflag("-std=c99")
-add_cxxflag("-std=c++11")
-
-# Error on non-ISO C.
-add_cflag("-pedantic-errors")
 
 option(WARNINGS "Enable additional compiler warnings" ON)
 if(WARNINGS)
+  # Set standard version for compiler.
+  add_cflag("-std=gnu99")
+  add_cxxflag("-std=c++98")
+
```

Comment:
We also compile the code as C++ to make sure it's C++ compatible. Reasons:
- Additional C++-only tooling becomes available (static analysers).
- A few checks performed by C++ compilers (enum compatibility, void pointer casts) are useful for us.